src/odemis/driver/test/actuator_test.py

```diff
@@ -1274,6 +1275,225 @@ def tearDown(self):
         self.dev_cycle.terminate()
         super(TestRotationActuator, self).tearDown()
 
+CONFIG_5DOF = {"name": "5DOF",
+               "role": "stage",
+               "ref_on_init": True,
+               "linear_speed": 0.001,  # m/s
+               "locator": "network:sn:MCS2-00001602",
+               "axes": {
+                   'x': {
+                       'range': [-3e-3, 3e-3],
+                       'unit': 'm',
+                   },
+                   'y': {
+                       'range': [-3e-3, 3e-3],
+                       'unit': 'm',
+                   },
+                   'z': {
+                       'range': [-3e-3, 3e-3],
+                       'unit': 'm',
+                   },
+                   'rx': {
+                       'range': [-0.785, 0.785],
+                       'unit': 'rad',
+                   },
+                   'rz': {
+                       'range': [-0.785, 0.785],
+                       'unit': 'rad',
+                   },
+               },
+               }
+TEST_NOHW = True
+if TEST_NOHW:
+    CONFIG_5DOF['locator'] = 'fake'
+
+CONFIG_3DOF = {"name": "3DOF",
+               "role": "stage",
+               "ref_on_init": True,
+               "locator": "network:sn:MCS2-00001604",
+               "speed": 0.1,
+               "accel": 0.001,
+               "hold_time": 1.0,
+               "axes": {
+                   'x': {
+                       'range': [-3e-3, 3e-3],
+                       'unit': 'm',
+                       'channel': 0,
+                   },
+                   'y': {
+                       'range': [-3e-3, 3e-3],
+                       'unit': 'm',
+                       'channel': 1,
+                   },
+                   'z': {
+                       'range': [-3e-3, 3e-3],
+                       'unit': 'm',
+                       'channel': 2,
+                   },
+               },
+               }
+
+if TEST_NOHW:
+    CONFIG_3DOF['locator'] = 'fake'
+
+class TestLinkedHeightActuator(unittest.TestCase):
```

Comment:
The test cases need to test more (corner) cases:
 * cancelling a move
 * cancelling a reference
 * reference stage with lens Z is active
 * Try to move Rx with lens Z active
 * Try to set lens Z to active with Rx set
 * Change MD_POS_COR => focus position should immediately be updated
 * Change focus DEACTIVE/ACTIVE => fails
 * Move lens near DEACTIVE, => focus should report it's DEACTIVE
 * Move focus to min/max of range => Check that ACTIVE is updated
 * On the edges of focus range, try to move relative a tiny bit too far => it should fail
 * Move focus to DEACTIVE => check that ACTIVE is not updated

BTW, for checking failures, do you know you can use a construct like `with self.assertRaise(ValueError):`? It's quite handy.